base_technical_features/models/basemodel_monkeypatch.py

```diff
@@ -1,33 +1,30 @@
 # -*- coding: utf-8 -*-
 # © 2016 Opener B.V. (<https://opener.am>)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
-from openerp import models, api
+from openerp import models
 
 
 class BaseModelMonkeyPatch(models.AbstractModel):
     _name = 'basemodel.monkeypatch'
 
-    def _register_hook(self, cr):
+    def _register_hook(self):
         if not hasattr(
                 models.BaseModel, 'base_technical_features_user_has_groups'):
 
             models.BaseModel.base_technical_features_user_has_groups = (
                 models.BaseModel.user_has_groups)
 
-            @api.cr_uid_context
-            def user_has_groups(self, cr, uid, groups, context=None):
+            def user_has_groups(self, groups):
                 """ Return True for users in the technical features group when
                 membership of the original group is checked, even if debug mode
                 is not enabled.
                 """
                 if ('base.group_no_one' in groups.split(',') and
-                    self.pool['res.users'].has_group(
-                        cr, uid,
+                    self.env.user.has_group(
                         'base_technical_features.group_technical_features')):
                     return True
-                return self.base_technical_features_user_has_groups(
-                    cr, uid, groups, context=context)
+                return self.base_technical_features_user_has_groups(groups)
 
             models.BaseModel.user_has_groups = user_has_groups
 
-        return super(BaseModelMonkeyPatch, self)._register_hook(cr)
+        return super(BaseModelMonkeyPatch, self)._register_hook()
```

Comment:
@StefanRijnhart I'm no expert in these matter but is there not now a way in 10 to extend BaseModel without resorting to a monkey patch?